### ISSUE TITLE
Implement `createIndexes` for unique indexes

### ISSUE THREAD

**Issue by rumyantseva** [FerretDB/FerretDB#2045]
### What should be done?

Support `unique` option for `createIndexes` command.

In the pgdb package, the `CreateIndexIfNotExists` function already accepts `pgdb.Index` struct and supports `unique` option. The only thing that needs to be done is implementation of the logic on the handler level.

### Where

`MsgCreateIndexes`

### Definition of done

- [x]  Implement the creation of unique indexes in the handler.
- [x] Compat tests: check that after creating a unique index `listIndexes` returns the same results for target and compat.
- [x] Compat tests: check that an attempt to insert a duplicate value in a field with a unique index will return the correct error. Bonus points: create a unique index with a few fields and check the same.
- [x] Unit tests: it's possible to check that creating a unique index causes the correct index definition to be set, see `TestCreateIndexIfNotExists`.
- [x] Check TODO in the code.
- [ ] Update supported commands documentation.

**Comment by AlekSi** [comment]
@rumyantseva Please use our Chore template for description and prepare this issue for estimation

**Comment by rumyantseva** [comment]
@adetunjii thank you! Assigned. As usual, feel free to ask any questions regarding the issue.

**Comment by rumyantseva** [comment]
@adetunjii hi there! Just wanted to check in regarding the task you volunteered for. Are you still able to work on it? If you need any assistance, please let me know. Thanks!

**Comment by adetunjii** [comment]
@rumyantseva Yes I'm still able to work on it. Had to go off for a while. I'm back now and I'll see to it soon

**Comment by w84thesun** [comment]
We have a TODO for extending error messages for non-`_id` unique indexes. To add support for these errors we have to get an index name from the error code and use indexes metadata to get the correct index name. We may not want to do this, considering the refactoring of the metadata package ahead.

There is another strange behaviour that I found. We have a collection with `Int32s` and we trying to create an index for not existing field, like in [this](https://github.com/FerretDB/FerretDB/pull/2814/files#diff-b2e4c964b500457dfe583f5f7f7138a8871544e5338b53bbc0d8a02721a9e46bR737) test case. MongoDB would return an index build error, while PG would just create the required index. I don't quite understand how we should proceed with it.

The third thing I found is that we have two errors with the same code `11000` but different names. For `insert` its name is `Location11000` and for `createIndexes` its `DuplicateKey` as shown in [TestCreateIndexesErrors/DuplicateValue](https://github.com/FerretDB/FerretDB/pull/2814/files#diff-1a32d052b5be5658e85db107ed6a7606715b733d5da378757ef0c7d734c00a71R49). Our errors stored in a map, so it wouldn't be possible to create 2 errors with the same code. The way out I see is to document that difference in dance but it needs approval as we have that discussion about the errors a while ago.
